src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -231,7 +231,7 @@ public void cxParseResults(ScanRequest request, File file) throws ExitThrowable
         try {
             ScanResults results = cxService.getReportContent(file, request.getFilter());
             resultsService.processResults(request, results, scanDetails);
-            if (flowProperties.isBreakBuild() && results != null && results.getXIssues() != null && !results.getXIssues().isEmpty()) {
+            if ((flowProperties.isBreakBuild() || request.isBreakBuildArgument()) && results != null && results.getXIssues() != null && !results.getXIssues().isEmpty()) {
```

Comment:
instead of null check chaines
Optional.ofNullable(results).map(ScanResults::getXIssues).map(xIssues->!x.Issues.isEmpty()).isPresent()